src/main/java/uk/gov/pay/api/service/PublicApiUriGenerator.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.pay.api.service;
+
+import javax.ws.rs.core.UriBuilder;
+import java.net.URI;
+
+public class PublicApiUriGenerator {
```

Comment:
how about initialising PublicApiUriGenerator with a baseUrl? looks like baseUrl is always the same.

Reply:
We now get Guice to pass the `PublicApiConfig` to the `PublicApiUriGenerator` constructor, which retrieves the `baseUrl` from it and stores it in a field